packages/@aws-cdk/aws-batch/lib/compute-environment.ts

```diff
@@ -0,0 +1,432 @@
+     *
+     * @default MANAGED
+     */
+    readonly type?: ComputeEnvironmentType;
+}
+
+/**
+ * Properties of a compute environment
+ */
+export interface IComputeEnvironment extends IResource {
+    /**
+     * The ARN of this compute environment
+     *
+     * @attribute
+     */
+    readonly computeEnvironmentArn: string;
+
+    /**
+     * The name of this compute environment
+     *
+     * @attribute
+     */
+    readonly computeEnvironmentName: string;
+}
+
+/**
+ * Batch Compute Environment
+ *
+ * Defines a batch compute environment to run batch jobs on.
+ */
+export class ComputeEnvironment extends Resource implements IComputeEnvironment {
+    /**
+     * Fetches an existing batch compute environment by its amazon resource name.
+     *
+     * @param scope
+     * @param id
+     * @param computeEnvironmentArn
+     */
+    public static fromComputeEnvironmentArn(scope: Construct, id: string, computeEnvironmentArn: string): IComputeEnvironment {
+        const stack = Stack.of(scope);
+        const computeEnvironmentName = stack.parseArn(computeEnvironmentArn).resource;
+
+        class Import extends Resource implements IComputeEnvironment {
+            public readonly computeEnvironmentArn = computeEnvironmentArn;
+            public readonly computeEnvironmentName = computeEnvironmentName;
+        }
+
+        return new Import(scope, id);
+    }
+
+    /**
+     * The ARN of this compute environment
+     *
+     * @attribute
+     */
+    public readonly computeEnvironmentArn: string;
+
+    /**
+     * The name of this compute environment
+     *
+     * @attribute
+     */
+    public readonly computeEnvironmentName: string;
+
+    constructor(scope: Construct, id: string, props: ComputeEnvironmentProps = {}) {
+        super(scope, id, {
+            physicalName: props.computeEnvironmentName,
+        });
+
+        this.validateProps(props);
+
+        let computeResources: CfnComputeEnvironment.ComputeResourcesProperty | undefined;
+
+        // Only allow compute resources to be set when using MANAGED type
+        if (props.computeResources && props.type === ComputeEnvironmentType.UNMANAGED) {
+            computeResources = {
+                allocationStrategy: props.allocationStrategy || AllocationStrategy.BEST_FIT,
+                bidPercentage: props.computeResources.bidPercentage,
+                desiredvCpus: props.computeResources.desiredvCpus,
+                ec2KeyPair: props.computeResources.ec2KeyPair,
+                imageId: props.computeResources.image ? props.computeResources.image.getImage(this).imageId : undefined,
+                instanceRole: props.computeResources.instanceRole ?
+                    props.computeResources.instanceRole.roleArn :
+                    new iam.Role(this, 'Resource-Role', {
+                        assumedBy: new iam.ServicePrincipal('batch.amazonaws.com'),
+                    }).roleArn,
```

Comment:
Hey. Ran into another hickup. This will fail, because Cloudformation expects an Instance Profile:
```
5/7 | 1:08:40 PM | CREATE_FAILED        | AWS::Batch::ComputeEnvironment | compute-env (computeenv5B70C65C)
    Operation failed, ComputeEnvironment went INVALID with error: 
    CLIENT_ERROR - Invalid IamInstanceProfile: arn:aws:iam::0123456789:role/batch-compute-env-stack-computeenvResourceRole92B0-Q4UIZBETE6B1
```

You will need to create a `new iam.CfnInstanceProfile`.

Reply:
Good eye, I fixed this up now. We needed to make sure to use the `service-role/AWSBatchServiceRole`.

I didn't need to end up using that L0 construct.